syntax = "proto2";

package WatchdogService;

enum Operation {
  ConnectRequest = 0;
  ConnectResponse = 1;
  PingRequest = 2;
  PingResponse = 3;
  ReconnectRequest = 4;
  ReconnectResponse = 5;
  ShutdownRequest = 6;
}

message ConnectRequestData {
  required uint32 identifier = 1;
}

enum ResponseCode {
  Success = 0;
  InvalidConnectionState = 1;
  NotServiceIdentifier = 2;
  ServiceNotExists = 3;
  Unknown = 4;
}

message ConnectResponseData {
  required ResponseCode responseCode = 1 [default = Unknown];
  optional uint32 sequenceCode = 2;
}

message PingRequestData {
  required uint32 sequenceCode = 1;
}

message PingResponseData {
  required uint32 sequenceCode = 1;
}

message ReconnectRequestData {
  required uint32 identifier = 1;
}

message ReconnectResponseData {
  required ResponseCode responseCode = 1 [default = Unknown];
  optional uint32 sequenceCode = 2;
}

message ShutdownRequestData {
  required uint32 identifier = 1;
}